### ISSUE TITLE
Kodi Fails to build

### ISSUE THREAD

**Issue by acjones8** [InBetweenNames/gentooLTO#783]
Hi everyone. Kodi's new version failed to build with this error:
```
    lto1: internal compiler error: Segmentation fault
0xd4d223 internal_error(char const*, ...)
	???:0
0xe3b7f6 bp_unpack_string(data_in*, bitpack_d*)
	???:0
0x1292278 cl_optimization_stream_in(data_in*, bitpack_d*, cl_optimization*)
	???:0
0xe382d2 streamer_read_tree_bitfields(lto_input_block*, data_in*, tree_node*)
	???:0
0xe35094 lto_input_tree_1(lto_input_block*, data_in*, LTO_tags, unsigned int)
	???:0
0xe39fce lto_input_scc(lto_input_block*, data_in*, unsigned int*, unsigned int*, bool)
	???:0
0xe2d621 read_cgraph_and_symbols(unsigned int, char const**)
	???:0
0x128f8a6 lto_main()
	???:0
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://bugs.gentoo.org/> for instructions.
lto-wrapper: fatal error: /usr/bin/x86_64-pc-linux-gnu-g++ returned 1 exit status
compilation terminated.
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
 ERROR: media-tv/kodi-19.1-r1::gentoo failed (compile phase):
 ninja -v -j4 -l0 all failed

 Call stack:
     ebuild.sh, line  127:  Called src_compile
   environment, line 6109:  Called cmake_src_compile 'all'
  environment, line 3312:  Called cmake_build 'all'
   environment, line 3281:  Called eninja 'all'
   environment, line 3972:  Called die
The specific snippet of code:
     "$@" || die "${nonfatal_args[@]}" "${*} failed"

  If you need support, post the output of `emerge --info '=media-tv/kodi-19.1-r1::gentoo'`,
 the complete build log and the output of `emerge -pqv '=media-tv/kodi-19.1-r1::gentoo'`.
The complete build log is located at '/var/tmp/portage/media-tv/kodi-19.1-r1/temp/build.log'.
 The ebuild environment file is located at '/var/tmp/portage/media-tv/kodi-19.1-r1/temp/environment'.
  Working directory: '/var/tmp/portage/media-tv/kodi-19.1-r1/work/kodi-19.1_build'
 S: '/var/tmp/portage/media-tv/kodi-19.1-r1/work/xbmc-19.1'
```

It looks like an issue with LTO. I'll try building it without it later today, hopefully we don't have to ditch LTO completely.

**Comment by acjones8** [comment]
Alright, so unfortunately building it with *FLAGS-=-flto* in lto.conf, but unfortunately that didn't fix it. I'm certain that this must be an issue with LTO, it's too suspicious to me that it would fail on the linking step like this, but I'm not quite sure where to go from here. I dug up [this thread](https://forum.kodi.tv/showthread.php?tid=363084) that talks about issues with LTO and Kodi on Clang, so it could be that it's broken for GCC too, though disabling LTO should fix that...

**Comment by acjones8** [comment]
Nevermind: after running `lto-rebuild -r` and applying the fix from #765, Kodi builds just fine.  This was just a symptom of a wider issue.
